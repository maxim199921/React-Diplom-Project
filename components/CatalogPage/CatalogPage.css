.CatalogPage {
    margin-top: 120px;
    background-color: #efefef;
    color: white;
}

.CatalogPage-container-shad{
    box-shadow: 0 0 10px 5px rgba(22,35,23,.5);
}

.CatalogPage-wrapper-shad {
    padding-top: 20px;
    padding-bottom: 20px;
    box-shadow: 0 15px 10px -7px rgba(22,35,23,.5);
    z-index: 1;
}

.CatalogPage-wrapper-shad-Filter {
    box-shadow: 5px 14px 10px 0 rgba(22,35,23,.5);
    z-index: 1;
}

.CatalogPage-List {
    background-color: #efefef;
}

.CatalogPage-Filter {
    background-color: #5dbbc1;
    padding-top: 50px;
    padding-left: 70px;
    min-height: 100vh;
}

.btn-primary-buy {
    margin-top: 150px;
    width: 60%;
}

.CatalogPage-Text {
    background-color: #5dbbc1;
}

.CatalogPage-Text-Loading {
    height: 100vh;
    margin-top: 20px;
    font-size: 42px;
    line-height: 50px;
}

.CatalogPage .radio input {
    position: absolute;
    z-index: -1;
    opacity: 0;
    margin: 10px 0 0 7px;
}
.CatalogPage .radio__text {
    position: relative;
    padding: 0 0 0 35px;
    cursor: pointer;
    font-weight: bold;
}
.CatalogPage .radio__text:before {
    content: '';
    position: absolute;
    top: 4px;
    left: 0;
    width: 18px;
    height: 18px;
    border: 1px solid #CDD1DA;
    border-radius: 50%;
    background: #FFF;
}
.CatalogPage .radio__text:after {
    content: '';
    position: absolute;
    top: 8px;
    left: 4px;
    width: 10px;
    height: 10px;
    border-radius: 50%;
    background: #130e27;
    box-shadow: inset 0 1px 1px rgba(0,0,0,.5);
    opacity: 0;
    transition: .2s;
}
.CatalogPage .radio input:checked + .radio__text:after {
    opacity: 0.7;
}
.CatalogPage .radio input:checked + .radio__text:before {
    box-shadow: 0 0 15px 5px rgba(22,35,23,.5);
}

.CatalogPage-header {
    padding-top: 15px;
    padding-bottom: 15px;
    margin-left: 15px;
    font-size: 36px;
    line-height: 40px;
    font-weight: bold;
}

.CatalogPage .btn-primary {
    background-color: #130e27 !important;
    border-color: #130e27 !important;
    box-shadow: #130e27 !important;
    font-weight: bold;
}

.CatalogPage-header-btn {
    margin-right: 5px;
    margin-left: 50%;
}

.CatalogPage-header-search {
    width: 100px;
}

@media (max-width: 991px) {
    .CatalogPage-Filter {
        padding-left: 30px;
    }
    .btn-primary-buy {
        margin-top: 40px;
        width: 90%;
    }
}

@media (max-width: 767px) {
    .CatalogPage-header-btn {
        margin-left: 30%;
    }
    .CatalogPage-Filter {
        padding-left: 20px;
    }
    .btn-primary-buy {
        margin-top: 40px;
        width: 100%;
    }
}


@media (max-width: 575px) {
    .CatalogPage-header-btn {
        margin-left: 0;
        margin-bottom: 25px;
    }
    .CatalogPage-header-search {
        width: 150px;
        margin-bottom: 25px;
    }
    .CatalogPage-Text {
        text-align: center;
    }
    .CatalogPage-Filter {
        display: none;
    }
}

.CatalogPage-Filter p {
    margin: 0;
}

.noFountError {
    color: orange;
    font-weight: bold;
    margin: 0 auto;
    margin-top: 50px;
}


.loader-container {
    font-size: 10px;
    position: absolute;
    width: 100px;
    height: 100px;
    display: inline-block;
    text-align: center;
    border-collapse: collapse;
    vertical-align: top;
    top: -3px;
}

.loader-1 {
    width: 50px;
    height: 50px;
    top: 50%;
    margin: -25px auto 0;
    border-radius: 50%;
    background: #ffffff;
    background: -moz-linear-gradient(left, #130e27 10%, cadetblue 50%);
    background: -webkit-linear-gradient(left, #130e27 10%, cadetblue 50%);
    background: -o-linear-gradient(left, #130e27 10%, cadetblue 50%);
    background: -ms-linear-gradient(left, #130e27 10%, cadetblue 50%);
    background: linear-gradient(to right, #130e27 10%, cadetblue 50%);
    position: relative;
    -webkit-animation: loader-1 1.5s infinite linear;
    animation: loader-1 1.5s infinite linear;
    -webkit-transform: translateZ(0);
    -ms-transform: translateZ(0);
    transform: translateZ(0);
}

.loader-1:before {
    content: '';
    width: 25px;
    height: 25px;
    position: absolute;
    top: 0;
    left: 0;
    background: #ffffff;
    border-radius: 100% 0 0 0;
}

.loader-1:after {
    content: '';
    width: 40px;
    height: 40px;
    border-radius: 50%;
    background: #ffffff;
    margin: auto;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
}

@-webkit-keyframes loader-1 {
    0% {
        -webkit-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

@keyframes loader-1 {
    0% {
        -webkit-transform: rotate(0deg);
        transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

