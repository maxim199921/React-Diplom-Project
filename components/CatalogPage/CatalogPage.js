import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {NavLink} from 'react-router-dom';
import {loadData, setPriceListFromLocalStorage, filterLaptor, filterPhone, filterTablet, filterAll, search} from '../../redux/ActionCreaters/CatalogActions'
import CatalogList from './CatalogList/CatalogList';
import './CatalogPage.css';

class CatalogPage extends React.PureComponent {

    static propTypes = {
        isLoaded: PropTypes.bool.isRequired,
        checkedRadio: PropTypes.number.isRequired,
        foundNull: PropTypes.bool.isRequired,
        priceList: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                type: PropTypes.string.isRequired,
                img: PropTypes.string.isRequired,
                model: PropTypes.string.isRequired,
                price: PropTypes.string.isRequired,
                description: PropTypes.string.isRequired,
                status: PropTypes.bool.isRequired,
            })
        ),
        startPriceList: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                type: PropTypes.string.isRequired,
                img: PropTypes.string.isRequired,
                model: PropTypes.string.isRequired,
                price: PropTypes.string.isRequired,
                description: PropTypes.string.isRequired,
                status: PropTypes.bool.isRequired,
            })
        ),
    };

    state = {
        searchValue: null,
    };

    componentDidMount = () => {
        if (localStorage.startPriceList === undefined){
            this.props.dispatch(loadData());
        } else {
            this.props.dispatch( setPriceListFromLocalStorage(JSON.parse(localStorage.startPriceList)) );
        }
        this.scrollTop();

    };

    componentDidUpdate = () => {
        localStorage['startPriceList'] = JSON.stringify(this.props.startPriceList);
    };

    scrollTop = () => {
        window.scrollTo(0, 0);
    };

    setSearchValue = (evt) => {
        this.setState({searchValue: evt.target.value});
    };
    searchFilterPriceList = () => {
        this.props.dispatch( search(this.state.searchValue) )
    };
    searchKeyPressEnter = (event) => {
        if(event.key === 'Enter'){
            event.preventDefault();
            this.props.dispatch( search(this.state.searchValue) )
        }
    };


    showLaptor = () => {this.props.dispatch( filterLaptor('Laptop') )};
    showPhone = () => {this.props.dispatch( filterPhone('Phone') )};
    showTablet = () => {this.props.dispatch( filterTablet('Tablet') )};
    showAll = () => {this.props.dispatch( filterAll() )};

    render() {

        let catalogList;
        this.props.foundNull
            ? catalogList = <h1 className="noFountError">...no products found try again</h1>
            : catalogList = this.props.priceList.map(item =>
                <CatalogList key={item.id}
                              info={item}/>
             );

        return (

            <div className="CatalogPage">
                <div className="container CatalogPage-container-shad">
                    <div className="row">
                        <div className="col-12 CatalogPage-Text">
                            <div className="row">
                                <div className="col-12 col-sm-3">
                                    <h2 className="CatalogPage-header">Catalog</h2>
                                </div>
                                <div className="col-12 col-sm-9 align-self-center CatalogPage-wrapper-shad">
                                    <form className="form-inline">
                                            <input type="button" className="btn btn-primary CatalogPage-header-btn" value="GO"
                                            onClick={this.searchFilterPriceList}/>
                                            <input type="text" className="form-control CatalogPage-header-search" placeholder="search"
                                            onChange={this.setSearchValue}
                                            onKeyPress={this.searchKeyPressEnter}/>
                                    </form>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className="row">
                        <div className="col-12 col-sm-3 CatalogPage-Filter CatalogPage-wrapper-shad-Filter">
                            <label className="radio" htmlFor="SHOW_ALL_RADIO">
                                <input id="SHOW_ALL_RADIO" name="filter" type="radio" value="SHOW ALL"  onChange={this.showAll}
                                       checked={this.props.checkedRadio === 0}/>
                                <div className="radio__text">SHOW ALL</div>
                            </label>
                            <label className="radio" htmlFor="LAPTOR_RADIO">
                                <input id="LAPTOR_RADIO" name="filter" type="radio" value="LAPTOR"  onChange={this.showLaptor}
                                       checked={this.props.checkedRadio === 1}/>
                                <div className="radio__text">LAPTOP</div>
                            </label><br/>
                            <label className="radio" htmlFor="PHONE_RADIO">
                                <input id="PHONE_RADIO" name="filter" type="radio" value="PHONE"  onChange={this.showPhone}
                                       checked={this.props.checkedRadio === 2}/>
                                <div className="radio__text">PHONE</div>
                            </label><br/>
                            <label className="radio" htmlFor="TABLET_RADIO">
                                <input id="TABLET_RADIO" name="TABLET_RADIO" type="radio" value="TABLET" onChange={this.showTablet}
                                       checked={this.props.checkedRadio === 3}/>
                                <div className="radio__text">TABLET</div>
                            </label><br/>
                            <NavLink to="/basketpage" className="btn btn-primary btn-primary-buy">Buy</NavLink>
                        </div>
                        <div className="col-12 col-sm-9 CatalogPage-List">
                            {
                                (this.props.isLoaded)?
                                <div className="row">
                                    {catalogList}
                                </div>
                                :
                                <div className="CatalogPage-Loader">
                                    <div className="container">
                                        <div className="row">
                                            <div className="col-12">
                                                <h1 className="CatalogPage-Text-Loading">...Loading
                                                    <div className="loader-container">
                                                        <div className="loader-1"></div>
                                                    </div>
                                                </h1>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            }
                        </div>
                    </div>
                </div>
            </div>

        );

    }

}

const mapStateToProps = function (state) {
    return {
        isLoaded: state.catalogList.isLoaded,
        priceList: state.catalogList.priceList,
        checkedRadio: state.catalogList.checkedRadio,
        foundNull: state.catalogList.foundNull,
        startPriceList: state.catalogList.startPriceList,
    };
};


export default connect(mapStateToProps)(CatalogPage);
