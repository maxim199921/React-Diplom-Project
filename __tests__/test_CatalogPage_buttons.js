import React from 'react';
import renderer from 'react-test-renderer';
import CatalogPage from '../components/CatalogPage/CatalogPage';
import {HashRouter, hashHistory} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from '../redux/configureStore';
const store = configureStore({});


test('работа кнопок каталога', () => {


    const component = renderer.create(
        <Provider store={store}>
            <HashRouter history={hashHistory}>
                <CatalogPage/>
            </HashRouter>
        </Provider>

    );

    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const searchButton = component.root.find( el => el.props.type ==="text" );

    searchButton.props.onKeyPress('Enter');

    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const goShowAllButton = component.root.find( el => el.props.value ==="SHOW ALL" );

    goShowAllButton.props.onChange();

    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const goShowLAPTORButton = component.root.find( el => el.props.value ==="LAPTOR" );

    goShowLAPTORButton.props.onChange();

    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const goShowPHONEButton = component.root.find( el => el.props.value ==="PHONE" );

    goShowPHONEButton.props.onChange();

    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const goShowTABLETButton = component.root.find( el => el.props.value ==="TABLET" );

    goShowTABLETButton.props.onChange();

    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
});