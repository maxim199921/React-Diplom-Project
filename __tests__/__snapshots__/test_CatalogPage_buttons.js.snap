// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа кнопок каталога 1`] = `
<div
  className="CatalogPage"
>
  <div
    className="container CatalogPage-container-shad"
  >
    <div
      className="row"
    >
      <div
        className="col-12 CatalogPage-Text"
      >
        <div
          className="row"
        >
          <div
            className="col-12 col-sm-3"
          >
            <h2
              className="CatalogPage-header"
            >
              Catalog
            </h2>
          </div>
          <div
            className="col-12 col-sm-9 align-self-center CatalogPage-wrapper-shad"
          >
            <form
              className="form-inline"
            >
              <input
                className="btn btn-primary CatalogPage-header-btn"
                onClick={[Function]}
                type="button"
                value="GO"
              />
              <input
                className="form-control CatalogPage-header-search"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="search"
                type="text"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-3 CatalogPage-Filter CatalogPage-wrapper-shad-Filter"
      >
        <label
          className="radio"
          htmlFor="SHOW_ALL_RADIO"
        >
          <input
            checked={true}
            id="SHOW_ALL_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="SHOW ALL"
          />
          <div
            className="radio__text"
          >
            SHOW ALL
          </div>
        </label>
        <label
          className="radio"
          htmlFor="LAPTOR_RADIO"
        >
          <input
            checked={false}
            id="LAPTOR_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="LAPTOR"
          />
          <div
            className="radio__text"
          >
            LAPTOP
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="PHONE_RADIO"
        >
          <input
            checked={false}
            id="PHONE_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="PHONE"
          />
          <div
            className="radio__text"
          >
            PHONE
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="TABLET_RADIO"
        >
          <input
            checked={false}
            id="TABLET_RADIO"
            name="TABLET_RADIO"
            onChange={[Function]}
            type="radio"
            value="TABLET"
          />
          <div
            className="radio__text"
          >
            TABLET
          </div>
        </label>
        <br />
        <a
          aria-current={null}
          className="btn btn-primary btn-primary-buy"
          href="#/basketpage"
          onClick={[Function]}
        >
          Buy
        </a>
      </div>
      <div
        className="col-12 col-sm-9 CatalogPage-List"
      >
        <div
          className="CatalogPage-Loader"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <h1
                  className="CatalogPage-Text-Loading"
                >
                  ...Loading
                  <div
                    className="loader-container"
                  >
                    <div
                      className="loader-1"
                    />
                  </div>
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`работа кнопок каталога 2`] = `
<div
  className="CatalogPage"
>
  <div
    className="container CatalogPage-container-shad"
  >
    <div
      className="row"
    >
      <div
        className="col-12 CatalogPage-Text"
      >
        <div
          className="row"
        >
          <div
            className="col-12 col-sm-3"
          >
            <h2
              className="CatalogPage-header"
            >
              Catalog
            </h2>
          </div>
          <div
            className="col-12 col-sm-9 align-self-center CatalogPage-wrapper-shad"
          >
            <form
              className="form-inline"
            >
              <input
                className="btn btn-primary CatalogPage-header-btn"
                onClick={[Function]}
                type="button"
                value="GO"
              />
              <input
                className="form-control CatalogPage-header-search"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="search"
                type="text"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-3 CatalogPage-Filter CatalogPage-wrapper-shad-Filter"
      >
        <label
          className="radio"
          htmlFor="SHOW_ALL_RADIO"
        >
          <input
            checked={true}
            id="SHOW_ALL_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="SHOW ALL"
          />
          <div
            className="radio__text"
          >
            SHOW ALL
          </div>
        </label>
        <label
          className="radio"
          htmlFor="LAPTOR_RADIO"
        >
          <input
            checked={false}
            id="LAPTOR_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="LAPTOR"
          />
          <div
            className="radio__text"
          >
            LAPTOP
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="PHONE_RADIO"
        >
          <input
            checked={false}
            id="PHONE_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="PHONE"
          />
          <div
            className="radio__text"
          >
            PHONE
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="TABLET_RADIO"
        >
          <input
            checked={false}
            id="TABLET_RADIO"
            name="TABLET_RADIO"
            onChange={[Function]}
            type="radio"
            value="TABLET"
          />
          <div
            className="radio__text"
          >
            TABLET
          </div>
        </label>
        <br />
        <a
          aria-current={null}
          className="btn btn-primary btn-primary-buy"
          href="#/basketpage"
          onClick={[Function]}
        >
          Buy
        </a>
      </div>
      <div
        className="col-12 col-sm-9 CatalogPage-List"
      >
        <div
          className="CatalogPage-Loader"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <h1
                  className="CatalogPage-Text-Loading"
                >
                  ...Loading
                  <div
                    className="loader-container"
                  >
                    <div
                      className="loader-1"
                    />
                  </div>
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`работа кнопок каталога 3`] = `
<div
  className="CatalogPage"
>
  <div
    className="container CatalogPage-container-shad"
  >
    <div
      className="row"
    >
      <div
        className="col-12 CatalogPage-Text"
      >
        <div
          className="row"
        >
          <div
            className="col-12 col-sm-3"
          >
            <h2
              className="CatalogPage-header"
            >
              Catalog
            </h2>
          </div>
          <div
            className="col-12 col-sm-9 align-self-center CatalogPage-wrapper-shad"
          >
            <form
              className="form-inline"
            >
              <input
                className="btn btn-primary CatalogPage-header-btn"
                onClick={[Function]}
                type="button"
                value="GO"
              />
              <input
                className="form-control CatalogPage-header-search"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="search"
                type="text"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-3 CatalogPage-Filter CatalogPage-wrapper-shad-Filter"
      >
        <label
          className="radio"
          htmlFor="SHOW_ALL_RADIO"
        >
          <input
            checked={true}
            id="SHOW_ALL_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="SHOW ALL"
          />
          <div
            className="radio__text"
          >
            SHOW ALL
          </div>
        </label>
        <label
          className="radio"
          htmlFor="LAPTOR_RADIO"
        >
          <input
            checked={false}
            id="LAPTOR_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="LAPTOR"
          />
          <div
            className="radio__text"
          >
            LAPTOP
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="PHONE_RADIO"
        >
          <input
            checked={false}
            id="PHONE_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="PHONE"
          />
          <div
            className="radio__text"
          >
            PHONE
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="TABLET_RADIO"
        >
          <input
            checked={false}
            id="TABLET_RADIO"
            name="TABLET_RADIO"
            onChange={[Function]}
            type="radio"
            value="TABLET"
          />
          <div
            className="radio__text"
          >
            TABLET
          </div>
        </label>
        <br />
        <a
          aria-current={null}
          className="btn btn-primary btn-primary-buy"
          href="#/basketpage"
          onClick={[Function]}
        >
          Buy
        </a>
      </div>
      <div
        className="col-12 col-sm-9 CatalogPage-List"
      >
        <div
          className="CatalogPage-Loader"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <h1
                  className="CatalogPage-Text-Loading"
                >
                  ...Loading
                  <div
                    className="loader-container"
                  >
                    <div
                      className="loader-1"
                    />
                  </div>
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`работа кнопок каталога 4`] = `
<div
  className="CatalogPage"
>
  <div
    className="container CatalogPage-container-shad"
  >
    <div
      className="row"
    >
      <div
        className="col-12 CatalogPage-Text"
      >
        <div
          className="row"
        >
          <div
            className="col-12 col-sm-3"
          >
            <h2
              className="CatalogPage-header"
            >
              Catalog
            </h2>
          </div>
          <div
            className="col-12 col-sm-9 align-self-center CatalogPage-wrapper-shad"
          >
            <form
              className="form-inline"
            >
              <input
                className="btn btn-primary CatalogPage-header-btn"
                onClick={[Function]}
                type="button"
                value="GO"
              />
              <input
                className="form-control CatalogPage-header-search"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="search"
                type="text"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-3 CatalogPage-Filter CatalogPage-wrapper-shad-Filter"
      >
        <label
          className="radio"
          htmlFor="SHOW_ALL_RADIO"
        >
          <input
            checked={false}
            id="SHOW_ALL_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="SHOW ALL"
          />
          <div
            className="radio__text"
          >
            SHOW ALL
          </div>
        </label>
        <label
          className="radio"
          htmlFor="LAPTOR_RADIO"
        >
          <input
            checked={true}
            id="LAPTOR_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="LAPTOR"
          />
          <div
            className="radio__text"
          >
            LAPTOP
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="PHONE_RADIO"
        >
          <input
            checked={false}
            id="PHONE_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="PHONE"
          />
          <div
            className="radio__text"
          >
            PHONE
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="TABLET_RADIO"
        >
          <input
            checked={false}
            id="TABLET_RADIO"
            name="TABLET_RADIO"
            onChange={[Function]}
            type="radio"
            value="TABLET"
          />
          <div
            className="radio__text"
          >
            TABLET
          </div>
        </label>
        <br />
        <a
          aria-current={null}
          className="btn btn-primary btn-primary-buy"
          href="#/basketpage"
          onClick={[Function]}
        >
          Buy
        </a>
      </div>
      <div
        className="col-12 col-sm-9 CatalogPage-List"
      >
        <div
          className="CatalogPage-Loader"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <h1
                  className="CatalogPage-Text-Loading"
                >
                  ...Loading
                  <div
                    className="loader-container"
                  >
                    <div
                      className="loader-1"
                    />
                  </div>
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`работа кнопок каталога 5`] = `
<div
  className="CatalogPage"
>
  <div
    className="container CatalogPage-container-shad"
  >
    <div
      className="row"
    >
      <div
        className="col-12 CatalogPage-Text"
      >
        <div
          className="row"
        >
          <div
            className="col-12 col-sm-3"
          >
            <h2
              className="CatalogPage-header"
            >
              Catalog
            </h2>
          </div>
          <div
            className="col-12 col-sm-9 align-self-center CatalogPage-wrapper-shad"
          >
            <form
              className="form-inline"
            >
              <input
                className="btn btn-primary CatalogPage-header-btn"
                onClick={[Function]}
                type="button"
                value="GO"
              />
              <input
                className="form-control CatalogPage-header-search"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="search"
                type="text"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-3 CatalogPage-Filter CatalogPage-wrapper-shad-Filter"
      >
        <label
          className="radio"
          htmlFor="SHOW_ALL_RADIO"
        >
          <input
            checked={false}
            id="SHOW_ALL_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="SHOW ALL"
          />
          <div
            className="radio__text"
          >
            SHOW ALL
          </div>
        </label>
        <label
          className="radio"
          htmlFor="LAPTOR_RADIO"
        >
          <input
            checked={false}
            id="LAPTOR_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="LAPTOR"
          />
          <div
            className="radio__text"
          >
            LAPTOP
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="PHONE_RADIO"
        >
          <input
            checked={true}
            id="PHONE_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="PHONE"
          />
          <div
            className="radio__text"
          >
            PHONE
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="TABLET_RADIO"
        >
          <input
            checked={false}
            id="TABLET_RADIO"
            name="TABLET_RADIO"
            onChange={[Function]}
            type="radio"
            value="TABLET"
          />
          <div
            className="radio__text"
          >
            TABLET
          </div>
        </label>
        <br />
        <a
          aria-current={null}
          className="btn btn-primary btn-primary-buy"
          href="#/basketpage"
          onClick={[Function]}
        >
          Buy
        </a>
      </div>
      <div
        className="col-12 col-sm-9 CatalogPage-List"
      >
        <div
          className="CatalogPage-Loader"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <h1
                  className="CatalogPage-Text-Loading"
                >
                  ...Loading
                  <div
                    className="loader-container"
                  >
                    <div
                      className="loader-1"
                    />
                  </div>
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`работа кнопок каталога 6`] = `
<div
  className="CatalogPage"
>
  <div
    className="container CatalogPage-container-shad"
  >
    <div
      className="row"
    >
      <div
        className="col-12 CatalogPage-Text"
      >
        <div
          className="row"
        >
          <div
            className="col-12 col-sm-3"
          >
            <h2
              className="CatalogPage-header"
            >
              Catalog
            </h2>
          </div>
          <div
            className="col-12 col-sm-9 align-self-center CatalogPage-wrapper-shad"
          >
            <form
              className="form-inline"
            >
              <input
                className="btn btn-primary CatalogPage-header-btn"
                onClick={[Function]}
                type="button"
                value="GO"
              />
              <input
                className="form-control CatalogPage-header-search"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="search"
                type="text"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-sm-3 CatalogPage-Filter CatalogPage-wrapper-shad-Filter"
      >
        <label
          className="radio"
          htmlFor="SHOW_ALL_RADIO"
        >
          <input
            checked={false}
            id="SHOW_ALL_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="SHOW ALL"
          />
          <div
            className="radio__text"
          >
            SHOW ALL
          </div>
        </label>
        <label
          className="radio"
          htmlFor="LAPTOR_RADIO"
        >
          <input
            checked={false}
            id="LAPTOR_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="LAPTOR"
          />
          <div
            className="radio__text"
          >
            LAPTOP
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="PHONE_RADIO"
        >
          <input
            checked={false}
            id="PHONE_RADIO"
            name="filter"
            onChange={[Function]}
            type="radio"
            value="PHONE"
          />
          <div
            className="radio__text"
          >
            PHONE
          </div>
        </label>
        <br />
        <label
          className="radio"
          htmlFor="TABLET_RADIO"
        >
          <input
            checked={true}
            id="TABLET_RADIO"
            name="TABLET_RADIO"
            onChange={[Function]}
            type="radio"
            value="TABLET"
          />
          <div
            className="radio__text"
          >
            TABLET
          </div>
        </label>
        <br />
        <a
          aria-current={null}
          className="btn btn-primary btn-primary-buy"
          href="#/basketpage"
          onClick={[Function]}
        >
          Buy
        </a>
      </div>
      <div
        className="col-12 col-sm-9 CatalogPage-List"
      >
        <div
          className="CatalogPage-Loader"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-12"
              >
                <h1
                  className="CatalogPage-Text-Loading"
                >
                  ...Loading
                  <div
                    className="loader-container"
                  >
                    <div
                      className="loader-1"
                    />
                  </div>
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
